packages/core/src/components/fields/MultiSchemaField.tsx

```diff
@@ -201,7 +201,8 @@ class AnyOfField<
     const rawErrors = get(errorSchema, ERRORS_KEY, []);
     const fieldErrorSchema = omit(errorSchema, [ERRORS_KEY]);
 
-    const option = retrievedOptions[selectedOption] || null;
+    const option =
+      selectedOption >= 0 ? retrievedOptions[selectedOption] || null : null;
```

Comment:
Why not just do this and revert L238?
```suggestion
    const option =
      selectedOption >= 0 ? retrievedOptions[selectedOption] || null : undefined;
```